code/campaignclassic/src/main/java/com/adobe/marketing/mobile/BasicTemplateNotificationBuilder.java

```diff
@@ -0,0 +1,139 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+package com.adobe.marketing.mobile;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.os.Build;
+import android.widget.RemoteViews;
+import androidx.annotation.NonNull;
+import androidx.core.app.NotificationCompat;
+import com.adobe.marketing.mobile.campaignclassic.R;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.util.StringUtils;
+
+public class BasicTemplateNotificationBuilder {
+    private static final String SELF_TAG = "BasicTemplateNotificationBuilder";
+
+    @NonNull static NotificationCompat.Builder construct(
+            final BasicPushTemplate pushTemplate, final Context context) {
+        Log.trace(
+                CampaignPushConstants.LOG_TAG,
+                SELF_TAG,
+                "Building a basic template push notification.");
+        final String channelId =
+                AEPPushNotificationBuilder.createChannelAndGetChannelID(pushTemplate, context);
+        final String packageName =
+                ServiceProvider.getInstance()
+                        .getAppContextService()
+                        .getApplication()
+                        .getPackageName();
+        final RemoteViews smallLayout =
+                new RemoteViews(packageName, R.layout.push_template_collapsed);
+        final RemoteViews expandedLayout =
+                new RemoteViews(packageName, R.layout.push_template_expanded);
+
+        // get push payload data
+        final String imageUrl = pushTemplate.getImageUrl();
+        if (!StringUtils.isNullOrEmpty(imageUrl)) {
+            final Bitmap image = CampaignPushUtils.download(imageUrl);
```

Comment:
Do you have a rough estimate from your testing on the time taken to download and build a 3 image notification?
Also would you be able to attach a couple screenshots of how each notification looks with this code.

Reply:
I added some test code to calculate the elapsed time to download three bitmaps, scale them to a smaller size, then create a carousel item and add each view to the view flipper. I ran this code a few times and see it consistently taking between 300-400 milliseconds.